rollup.config.js

```diff
@@ -0,0 +1,252 @@
+            rollupConfig(
+                'dist/' + outputFileName + '.esm.js',
+                'dist/' + outputFileName + '.min.js',
+                name,
+                'umd',
+                namedExports ? 'named' : 'default',
+                globals,
+                {
+                    forceAllTransforms: true,
+                    exclude: 'node_modules/@babel/runtime/**',
+                    minified: true,
+                    targets: {
+                        browsers: BROWSER_TARGET
+                    },
+                    transformRuntimeOptions: {absoluteRuntime: true}
+                },
+                {
+                    dedupe: dedupe
+                },
+                ['xhr2-cookies', 'web3-providers-ipc']
+            )
+        );
+    }
+
+    return mappedConfig;
+};
+
+/**
+ * Generates and returns a rollup config item
+ *
+ * @method rollupConfig
+ *
+ * @param {string} input
+ * @param {string} outputFile
+ * @param {string} outputName
+ * @param {string} outputType
+ * @param {string} exports
+ * @param {object} globals
+ * @param {object} babelOptions
+ * @param {object} resolverOptions
+ *
+ * @returns {{output: ({file, sourcemap, exports, globals, name, format}|{file, sourcemap, exports, format}), input: *, plugins: *[]}}
+ */
+function rollupConfig(input, outputFile, outputName, outputType, exports = 'auto', globals = {}, babelOptions = {}, resolverOptions = {}, external) {
+    return {
+        input: input,
+        external: external,
+        output: getOutput(outputFile, outputType, outputName, exports, globals),
+        plugins: getPlugins(outputType, babelOptions, resolverOptions)
+    };
+}
+
+/**
+ * Returns the correct output configuration
+ *
+ * @method getOutput
+ *
+ * @param {string} file
+ * @param {string} type
+ * @param {string} name
+ * @param {string} exports
+ * @param {object} globals
+ *
+ * @returns {{file: *, sourcemap: boolean, exports, format: *}|{file: *, sourcemap: boolean, exports: string, globals: *, name: *, format: string}}
+ */
+function getOutput(file, type, name, exports, globals) {
+    let config = {
+        exports: exports,
+        name: name,
+        file: file,
+        sourcemap: true,
+        format: type
+    };
+
+    if (type === 'umd') {
+        config.globals = globals;
+    }
+
+    return config;
+}
+
+/**
+ * Returns the correct plugins array
+ *
+ * @method getPlugins
+ *
+ * @param {string} type
+ * @param {object} babelOptions
+ * @param {object} resolverOptions
+ *
+ * @returns {Array}
+ */
+function getPlugins(type, babelOptions, resolverOptions) {
+    const babelPlugin = babel(
+        getBabelConfig(
+            babelOptions.exclude,
+            babelOptions.targets,
+            babelOptions.forceAllTransforms,
+            babelOptions.transformRuntimeOptions,
+            babelOptions.minified
+        )
+    );
+
+    switch (type) {
+        case 'umd':
+            return [
+                resolve(
```

Comment:
Am I missing something or only `umd` uses `resolve`? How do the others resolve dependencies?

Reply:
Because only the minified umd file does require the deps to be bundled in? It isn't required for cjs or esm because node or the used bundler does resolve and bundle those packages.

Reply:
Got it! Thanks